From bca8987afccb73978f63d0a1d85eefc9949e1982 Mon Sep 17 00:00:00 2001
From: Ryan Blue <blue@apache.org>
Date: Tue, 15 Sep 2015 10:33:05 -0700
Subject: [PATCH 105/140] KITE-1072: Fix SchemaTool flaky test.

This makes all of the tests use a randomly-generated table name, except
for the directory migration test that uses a table name set in the test
schema, "simple".
---
 .../kitesdk/data/hbase/tool/SchemaToolTest.java    |   13 +++++++++----
 1 file changed, 9 insertions(+), 4 deletions(-)

diff --git a/kite-data/kite-data-hbase/src/test/java/org/kitesdk/data/hbase/tool/SchemaToolTest.java b/kite-data/kite-data-hbase/src/test/java/org/kitesdk/data/hbase/tool/SchemaToolTest.java
index 1f1d3ea..e357f7d 100644
--- a/kite-data/kite-data-hbase/src/test/java/org/kitesdk/data/hbase/tool/SchemaToolTest.java
+++ b/kite-data/kite-data-hbase/src/test/java/org/kitesdk/data/hbase/tool/SchemaToolTest.java
@@ -18,6 +18,7 @@ package org.kitesdk.data.hbase.tool;
 import static org.junit.Assert.assertEquals;
 
 import java.io.File;
+import java.util.UUID;
 
 import org.apache.commons.io.FileUtils;
 import org.apache.hadoop.hbase.client.HBaseAdmin;
@@ -42,8 +43,8 @@ import org.kitesdk.data.hbase.testing.HBaseTestUtils;
  * Test of using schema tool to migrate schemas in a directory.
  */
 public class SchemaToolTest {
-  private static final String tableName = "simple";
   private static final String managedTableName = "managed_schemas";
+  private static String tableName;
   private static File simpleSchemaFile;
   private static String simpleSchema;
 
@@ -63,11 +64,11 @@ public class SchemaToolTest {
 
   @AfterClass
   public static void afterClass() throws Exception {
-    HBaseTestUtils.util.deleteTable(Bytes.toBytes(tableName));
   }
-  
+
   @Before
   public void before() throws Exception {
+    tableName = UUID.randomUUID().toString().substring(0, 8);
     tablePool = new HTablePool(HBaseTestUtils.getConf(), 10);
     manager = new DefaultSchemaManager(tablePool);
     tool = new SchemaTool(new HBaseAdmin(HBaseTestUtils.getConf()),
@@ -77,7 +78,7 @@ public class SchemaToolTest {
   @After
   public void after() throws Exception {
     tablePool.close();
-    HBaseTestUtils.util.truncateTable(Bytes.toBytes(tableName));
+    HBaseTestUtils.util.deleteTable(Bytes.toBytes(tableName));
     HBaseTestUtils.util.truncateTable(Bytes.toBytes(managedTableName));
   }
 
@@ -85,6 +86,10 @@ public class SchemaToolTest {
   public void testMigrateDirectory() throws Exception {
     tool.createOrMigrateSchemaDirectory("classpath:SchemaTool", true);
 
+    // the table name is set in the schema file, so use that table name to
+    // verify that creatOrMigrateSchemaDirectory worked
+    tableName = "simple";
+
     Dao<SimpleHBaseRecord> dao = new SpecificAvroDao<SimpleHBaseRecord>(
         tablePool, tableName, "SimpleHBaseRecord", manager);
     testBasicOperations(dao);
-- 
1.7.9.5

